Assets/Scripts/Battle/LuaEnemyEncounter.cs

```diff
@@ -95,6 +95,13 @@ internal class LuaEnemyEncounter : EnemyEncounter {
 
     private void PrepareWave() {
         DynValue nextWaves = script.GetVar("nextwaves");
+        if (nextWaves.Type.ToString() != "Table") {
```

Comment:
You can use the Type value (of type MoonSharp.Interpreter.DataType) to know the type of the DynValue without using ToString(). You may wanna use here DataType.Table and DataType.Nil.